src/DataFramesMeta.jl

```diff
@@ -583,6 +584,61 @@ macro transform(x, args...)
     esc(transform_helper(x, args...))
 end
 
+##############################################################################
+##
+## transform! & @transform!
+##
+##############################################################################
+
+
+function transform!_helper(x, args...)
+
+    t = (fun_to_vec(arg) for arg in args)
+
+    quote
+        $DataFrames.transform!($x, $(t...))
+    end
+end
+
+"""
+    @transform!(d, i...)
+
+Mutate `d` inplace to add additional columns or keys based on keyword arguments and return it.  No copies of existing columns are made, meaning modifications of the returned data frame may affect the input data frame as well.
+
+### Arguments
+
+* `d` : an `AbstractDataFrame`, or `GroupedDataFrame`
+* `i...` : keyword arguments defining new columns or keys
+
+### Returns
+
+* `::DataFrame`
+
+### Examples
+
+```jldoctest
+julia> using DataFramesMeta
+
+julia> df = DataFrame(A = 1:3, B = [2, 1, 2]);
+
+julia> df2 = @transform!(df, a = 2 * :A, x = :A .+ :B);
+
+julia> (df === df2) && df2
```

Comment:
How about dropping the `;` above so that the result is printed there (which is quite natural), and then here only do:
```suggestion
julia> df === df2
```

Same below for `@select!`.